administration/live-migration.adoc

```diff
@@ -0,0 +1,138 @@
+pages) at a lower rate than these are being copied.
+
+#### Completion time
+
+In some cases the virtual machine can write to different memory pages / disk
+blocks at a higher rate than these can be copied, which will prevent the
+migration process from completing in a reasonable amount of time.
+In this case, live migration will be aborted if it is running for a long
+perioud of time. The timeout is calculated base on the side of the VMI, it's
+memory and the ephemeral disks that are needed to be copied. The configurable
+parameter `completionTimeoutPerGiB`, which deafults to 800s is the time for GiB
+of data to wait for the migration to be completed before aborting it.
+A VMI with 8Gib of memory will time out after 6400 seconds.
+
+#### Progress timeout
+
+Live migration will also be aborted when it will be noticed that memory doesn't
+make any progress.
+The time to wait for live migration to make progress in transferring data is
+configurable by `progressTimeout` parameter, which defaults to 150s
+
+## Migration Status Reporting
+
+### Condition and migration method
+
+When starting a virtual machine instance, it has also been calculated whether
+the machine is live migratable. The result is being stored in the VMI
+`Conditions`. The calculation can be based on multiple parameters of the VMI,
+however, at the moment, the calculation is largely based on the `Access Mode`
+of the VMI volumes. Live migration is only permitted when the volume access
+mode is set to `ReadWriteMany`. Requests to migrate a non-LiveMigratable VMI
+will be rejected.
+
+The reported `Migration Method` is also being calculated during VMI start.
+`BlockMigration` indicates that some of the VMI disks require copying from the
+source to the destination. `LiveMigration` means that only the instance memory
+will be copied.
+
+```
+Status:
+  Conditions:
+    Status:                True
+    Type:                  LiveMigratable
+  Migration Method:  BlockMigration
+```
+
+### Migration Status
+
+The migration progress status is being reported in the VMI `Status`.
+Most importantly, it indicates whether the migration has been `Completed` and
+if it `Failed`
+Below is an example of a successful migration.
+
+```
+Migration State:
+    Completed:        true
+    End Timestamp:    2019-03-29T03:37:52Z
+    Migration Config:
+      Completion Timeout Per GiB:  800
+      Progress Timeout:             150
+    Migration UID:                  c64d4898-51d3-11e9-b370-525500d15501
+    Source Node:                    node02
+    Start Timestamp:                2019-03-29T04:02:47Z
+    Target Direct Migration Node Ports:
+      35001:                      0
+      41068:                      49152
+      38284:                      49153
+    Target Node:                  node01
+    Target Node Address:          10.128.0.46
+    Target Node Domain Detected:  true
+    Target Pod:                   virt-launcher-testvmimcbjgw6zrzcmp8wpddvztvzm7x2k6cjbdgktwv8tkq
```

Comment:
Do we need to expose this information?

Reply:
Yes, it's needed for the source pod to resolve the destination pod.